.github/workflows/conda.yml

```diff
@@ -4,7 +4,7 @@ on:
     branches:
       - main
     tags:
-      - "*"
+      - "*.*.*"
```

Comment:
Am curious why the additional `.*`s are needed now. Are there some cases we would like to filter out?

Reply:
This was just lifted from the `on.push.tags` block in the [release publisher workflow](https://github.com/dask/dask/blob/a747fd42a6f2d4e475579456ee0540333bf1fa97/.github/workflows/release-publish.yml#L3-L6) - made this change to be in line with what we consider an event for a pushed "release" tag, but don't think it should be needed for the rest of these changes and can drop if that makes sense